internal/dag/httpproxy_processor.go

```diff
@@ -318,11 +321,17 @@ func (p *HTTPProxyProcessor) computeHTTPProxy(proxy *contour_api_v1.HTTPProxy) {
 				}
 				if tls.ClientValidation.CertificateRevocationList != "" {
 					secretName := k8s.NamespacedNameFrom(tls.ClientValidation.CertificateRevocationList, k8s.DefaultNamespace(proxy.Namespace))
-					crl, err := p.source.LookupCRLSecret(secretName)
+					crl, err := p.source.LookupCRLSecret(secretName, proxy.Namespace)
 					if err != nil {
-						// CRL is missing or not configured.
-						validCond.AddErrorf(contour_api_v1.ConditionTypeTLSError, "ClientValidationInvalid",
-							"Spec.VirtualHost.TLS client validation is invalid: invalid CRL Secret %q: %s", secretName, err)
+						if _, ok := err.(DelegationNotPermittedError); ok {
+							validCond.AddErrorf(contour_api_v1.ConditionTypeTLSError, "DelegationNotPermitted",
+								"Spec.VirtualHost.TLS CRL Secret %q is invalid: %s", tls.ClientValidation.CACertificate, err)
```

Comment:
It'd be ideal to add some test coverage for this (valid delegation, invalid/missing delegation)

Reply:
Done!